src/Bicep.Core/TypeSystem/Az/ResourceTypeProviderFactory.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Azure.Bicep.Types;
+using Azure.Bicep.Types.Az;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Features;
+using Bicep.Core.Modules;
+using Bicep.Core.Registry.Oci;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.TypeSystem.Az;
+using Newtonsoft.Json;
+
+namespace Bicep.Core.TypeSystem
+{
+
+
+    public class ResourceTypeProviderFactory : IResourceTypeProviderFactory
+    {
+        private record Key(string Name, string Version);
+
+        private readonly Key BuiltInAzLoaderKey = new("az", IResourceTypeProvider.BuiltInVersion);
+        private readonly Dictionary<Key, IResourceTypeProvider> resourceTypeLoaders;
+
+        public ResourceTypeProviderFactory()
+        {
+            resourceTypeLoaders = new() {
+                {BuiltInAzLoaderKey, new AzResourceTypeProvider(new AzResourceTypeLoader(new AzTypeLoader()))},
```

Comment:
Can we avoid embedding "az" logic in the generic type provider factory?

Reply:
We can but what is the downside of pre-warming the cache with the embedded az provider?